From david-b@pacbell.net  Fri Sep  1 19:56:22 2006
Date: Fri, 01 Sep 2006 15:26:27 -0700
From: David Brownell <david-b@pacbell.net>
To: gregkh@kroah.com
Subject: gadgetfs patch for ep0out
Cc: linux-usb-devel@lists.sourceforge.net, s-hansen@cox.net
Message-Id: <20060901222627.38823195C5D@adsl-69-226-248-13.dsl.pltn13.pacbell.net>

From: Skip Hansen <s-hansen@cox.net>

For ep0out transfers (rare), be sure to copy the right data to userspace.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/inode.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/gadget/inode.c
+++ gregkh-2.6/drivers/usb/gadget/inode.c
@@ -1015,7 +1015,7 @@ ep0_read (struct file *fd, char __user *
 			else {
 				len = min (len, (size_t)dev->req->actual);
 // FIXME don't call this with the spinlock held ...
-				if (copy_to_user (buf, &dev->req->buf, len))
+				if (copy_to_user (buf, dev->req->buf, len))
 					retval = -EFAULT;
 				clean_req (dev->gadget->ep0, dev->req);
 				/* NOTE userspace can't yet choose to stall */
